elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchSchemaFactory.java

```diff
@@ -53,6 +60,24 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(ElasticsearchSchemaFactory.class);
 
+  // RestClient objects allocate system resources and are thread safe. Here, we
+  // cache them using a key derived from the hashCode()s of the parameters that
+  // define a RestClient.
+  private static Cache<Integer, RestClient> REST_CLIENTS = CacheBuilder.newBuilder()
```

Comment:
I’m surprised this isn’t final.

Reply:
@julianhyde in the latest push the member is final.  Also, these ES clients are dedicated to a single ES server so the consequences of a hash collision would indeed have been bad.  So the cache is no longer keyed on a hash.